src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -355,18 +419,8 @@ public void ProduceStart(bool immediate = true, bool appCompleted = false)
             var status = ReasonPhrases.ToStatus(StatusCode, ReasonPhrase);
 
             var responseHeader = CreateResponseHeader(status, appCompleted, ResponseHeaders);
-            SocketOutput.Write(
-                responseHeader.Item1,
-                (error, state) =>
-                {
-                    if (error != null)
-                    {
-                        Log.LogError("ProduceStart ", error);
-                    }
-                    ((IDisposable)state).Dispose();
-                },
-                responseHeader.Item2,
-                immediate: immediate);
+            SocketOutput.Write(responseHeader.Item1, immediate: immediate);
```

Comment:
I don't think ProduceStart should ever block waiting for writes to complete.  I know that most of the time `immediate = false`, but that's not always true.

I think we're going to have to make this async eventually anyway when FireOnStarting becomes async.